src/AWS.Deploy.CLI/Commands/DeployCommand.cs

```diff
@@ -194,6 +208,126 @@ public async Task ExecuteAsync(string stackName, bool saveCdkProject)
+                {
+                    object settingValue;
+                    try
+                    {
+                        switch (optionSetting.Type)
+                        {
+                            case OptionSettingValueType.String:
+                                settingValue = optionSettingValue;
+                                break;
+                            case OptionSettingValueType.Int:
+                                settingValue = int.Parse(optionSettingValue);
+                                break;
+                            case OptionSettingValueType.Bool:
+                                settingValue = bool.Parse(optionSettingValue);
+                                break;
+                            default:
+                                throw new InvalidOverrideValueException($"Invalid value {optionSettingValue} for option setting item {optionSettingJsonPath}");
+                        }
+                    }
+                    catch (Exception)
+                    {
+                        throw new InvalidOverrideValueException($"Invalid value {optionSettingValue} for option setting item {optionSettingJsonPath}");
+                    }
+                    
+                    optionSetting.SetValueOverride(settingValue);
+
+                    if (isPartOfDeploymentBundle)
+                        SetDeploymentBundleOptionSetting(recommendation, optionSetting.Id, settingValue);
+                }
+            }
+        }
+
+        private void SetDeploymentBundleOptionSetting(Recommendation recommendation, string optionSettingId, object settingValue)
+        {
+            switch (optionSettingId)
+            {
+                case "DockerExecutionDirectory":
+                    new DockerExecutionDirectoryCommand(_consoleUtilities).OverrideValue(recommendation, settingValue.ToString() ?? "");
+                    break;
+                case "DockerBuildArgs":
+                    new DockerBuildArgsCommand(_consoleUtilities).OverrideValue(recommendation, settingValue.ToString() ?? "");
+                    break;
+                case "DotnetBuildConfiguration":
+                    new DotnetPublishBuildConfigurationCommand(_consoleUtilities).Overridevalue(recommendation, settingValue.ToString() ?? "");
+                    break;
+                case "DotnetPublishArgs":
+                    new DotnetPublishArgsCommand(_consoleUtilities).OverrideValue(recommendation, settingValue.ToString() ?? "");
+                    break;
+                case "SelfContainedBuild":
+                    new DotnetPublishSelfContainedBuildCommand(_consoleUtilities).OverrideValue(recommendation, (bool)settingValue);
+                    break;
+                default:
+                    throw new OptionSettingItemDoesNotExistException($"The Option Setting Item { optionSettingId } does not exist.");
+            }
+        }

```

Comment:
I don't like what you are doing here. We don't need to differentiate between the deployment bundle option settings and the rest of the option settings. At the end, they are all option settings with Type Hints. As long as we are executing the type hints, we shouldn't have an issue. You don't need to manually execute the type hint commands based on a switch.

Reply:
There are 2 issues with trying to invoke TypeHintCommands for **_non-deployment-bundle option settings._**
* The TypeHintCommands are only available for top level option settings (ex. IAMRole, ECSCluster, VPC etc). However, when performing a deployment using a config file, we are only dealing with **_leaf option settings_** (ex. IAMRole.CreateNew, ECSCluster.NewClusterName etc.)
* These TypeHintCommands are only responsible for reading the input from the CLI and create a corresponding TypeHinResponse based on those values. But since we are only dealing with **_leaf option settings_**, we can directly set its value by calling in the `SetValueOverride` method. We will lose separation of concern if we amend the existing TypeHintCommands to facilitate the deployment using a config file (and will also complicate things in the process)

For the **_deployment-bundle option settings_**, the TypeHintCommands also perform some validation (since there are no validators specified as part of the recipe file). Due to this dependency I added the `OverrideValue` method that takes in the desired value, performs the **same validations** on it and finally sets it as part of the deployment bundle.